subsys/bluetooth/mesh/adv_legacy.c

```diff
@@ -53,7 +53,14 @@ static inline void adv_send(struct net_buf *buf)
 
 	adv_int = MAX(adv_int_min,
 		      BT_MESH_TRANSMIT_INT(BT_MESH_ADV(buf)->xmit));
-	duration = (BT_MESH_SCAN_WINDOW_MS +
+	/* Advertising enable could be delayed upto the end of scan window plus
+	 * one advertising interval. Subsequent advertisements require upto a
+	 * duration in multiples of adverising interval plus the advertising
+	 * random delay of upto 10 ms for each interval.
+	 * This is a calculation specific to Zephyr Bluetooth Low Energy
+	 * Controller when CONFIG_BT_CTLR_LOW_LAT=y used in nRF51x SoCs.
+	 */
+	duration = (BT_MESH_SCAN_WINDOW_MS + adv_int +
```

Comment:
What's causing this addition of the advertising interval? Does the controller now ensure that stopping and starting the advertiser does not produce any advertising intervals under the specified duration?

Reply:
In the Controller implementation for nRF51, each radio event prevents any scheduling related CPU use, i.e. inside a scan window event if advertising is enabled then advertising event is scheduled in intervals equivalent to the advertising interval until the event is outside the scan window duration. 

Hence, the first advertising radio event scheduled on air could be delayed upto the duration of the scan window plus one advertising interval.

(please request changes on this PR, to prevent it from being merged until we conclude on the current discussion).

Reply:
I see. Can this be wrapped inside relevant `IS_ENABLED()` checks, so we don't end up sending too many advertisements for the other modes? We're already sending too many for the normal case, to avoid sending none for some corner cases.